src/Tasks.UnitTests/Copy_Tests.cs

```diff
@@ -2464,5 +2472,22 @@ internal override void ErrorIfLinkFailedCheck()
         {
             base.ErrorIfLinkFailedCheck();
         }
+
+        private bool IsSymlinkingSupported(string sourceFile)
+        {
+            if (!NativeMethodsShared.IsWindows)
+            {
+                return true;
+            }
+
+            string symlinkFile = FileUtilities.GetTemporaryFile();
+            string errorMessage = null;
+            if (NativeMethodsShared.MakeSymbolicLink(symlinkFile, sourceFile, ref errorMessage))
+            {
+                File.Delete(symlinkFile);
+                return true;
+            }
+            return false;
+        }
```

Comment:
```suggestion
        private bool IsSymlinkingSupported(string sourceFile)
        {
            if (!NativeMethodsShared.IsWindows)
            {
                return true;
            }

            string symlinkFile = FileUtilities.GetTemporaryFile();
            string errorMessage = null;
            if (NativeMethodsShared.MakeSymbolicLink(symlinkFile, sourceFile, ref errorMessage))
            {
                File.Delete(symlinkFile);
                return true;
            }
            return false;
        }
```

Can it be the case when SymbolicLink can be created for one file but not for another file?
The solution with custom fact attribute may be as:
```
    public class SupportsSymbolicLinksFactAttribute : TheoryAttribute
    {
        public SupportsSymbolicLinksFactAttribute ()
        {
           if (!NativeMethodsShared.IsWindows)
            {
                return true;
            }

            string symlinkFile = FileUtilities.GetTemporaryFile();
            string sourceFile = FileUtilities.GetTemporaryFile();
            string errorMessage = null;
            if (NativeMethodsShared.MakeSymbolicLink(symlinkFile, sourceFile, ref errorMessage))
            {
                File.Delete(symlinkFile);
                return;
            }
            this.Skip = "This test requires support of symbolic link to be run.";
        }
    }

```

Reply:
Yes, this would be much cleaner. Let me try it..

Reply:
Seems to be working. Thank you!